### PR TITLE
NA - Added tag component

### PR DESCRIPTION
**Pull request by nagarajnetcentric**

### CODE REVIEW

File: apps/frontend/src/components/Tag/Tag.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+interface TagProps {
+  label: string;
+  isSelected?: boolean;
+}
+
+export const Tag = ({ label, isSelected }: TagProps) => {
+    const [selected, setSelected] = React.useState<boolean>(isSelected || false);
+
+  const classes = 'rounded-[40px] font-gellix text-[20px] leading-5 p-[30px] '.concat(
+    selected === true ? 'flex bg-midnight-blue text-white' : 'text-midnight-blue border border-midnight-blue hover:bg-midnight-blue hover:text-white',
```
Comment (adrielTosi) [inline apps/frontend/src/components/Tag/Tag.tsx:13]: I believe using the `clsx` library might improve this. 
```
clsx(commonClasses, {
  [selectedClasses]: selected,
  [notSelectedClasses: !selected
})
```

File: apps/frontend/src/components/Tag/Tag.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  );
+
+  const handleClick = ()=> {
+    setSelected(!selected);
+  }
+
+  return (
+    <div className={classes} onClick={handleClick}>
+      {label}
+      {selected && <img className='pl-[18px]' src={vector} alt="logo" />}
```
Comment (adrielTosi) [inline apps/frontend/src/components/Tag/Tag.tsx:23]: We should use `rem` instead of `px`

**Review by adrielTosi — CHANGES_REQUESTED** [review]

File: apps/frontend/src/components/Tag/Tag.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+interface TagProps {
+  label: string;
+  isSelected?: boolean;
+}
+
+export const Tag = ({ label, isSelected }: TagProps) => {
+    const [selected, setSelected] = React.useState<boolean>(isSelected || false);
+
+  const commonClasses = 'rounded-[2.5rem] font-gellix text-[1.25rem] leading-5 p-[1.875rem]';
```
Comment (adrielTosi) [inline apps/frontend/src/components/Tag/Tag.tsx:13]: Just a headsup, but not really a problem
`text-[1.25rem]` is tailwind `text-xl`

**Review by adrielTosi — APPROVED** [review]

**Review by alexiscoelho — CHANGES_REQUESTED** [review]

File: apps/frontend/src/components/Tag/Tag.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+interface TagProps {
+  label: string;
+  isSelected?: boolean;
+}
+
+export const Tag = ({ label, isSelected }: TagProps) => {
+    const [selected, setSelected] = React.useState<boolean>(isSelected || false);
+
+  const classes = 'rounded-[40px] font-gellix text-[20px] leading-5 p-[30px] '.concat(
```
[the remark this answers is not in the document]
Comment (nagarajnetcentric) [reply apps/frontend/src/components/Tag/Tag.tsx]: `text-xl` works but there is no class available for  `15px` and `30px` in padding so changed it to `px-[1.875rem] py-[0.938rem]`

File: apps/frontend/src/components/Tag/Tag.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    selected === true ? 'flex bg-midnight-blue text-white' : 'text-midnight-blue border border-midnight-blue hover:bg-midnight-blue hover:text-white',
+  );
+
+  const handleClick = ()=> {
+    setSelected(!selected);
+  }
+
+  return (
+    <div className={classes} onClick={handleClick}>
```
[the remark this answers is not in the document]
Comment (nagarajnetcentric) [reply apps/frontend/src/components/Tag/Tag.tsx]: Changed it to button

File: apps/frontend/src/components/Tag/Tag.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import vector from '../../assets/icons/vector.svg';
+
+interface TagProps {
+  label: string;
+  isSelected?: boolean;
+}
+
+export const Tag = ({ label, isSelected }: TagProps) => {
```
[the remark this answers is not in the document]
Comment (nagarajnetcentric) [reply apps/frontend/src/components/Tag/Tag.tsx]: Added onClick callback

File: apps/frontend/src/components/Tag/Tag.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    const [selected, setSelected] = React.useState<boolean>(isSelected || false);
+
+  const classes = 'rounded-[40px] font-gellix text-[20px] leading-5 p-[30px] '.concat(
+    selected === true ? 'flex bg-midnight-blue text-white' : 'text-midnight-blue border border-midnight-blue hover:bg-midnight-blue hover:text-white',
+  );
+
+  const handleClick = ()=> {
+    setSelected(!selected);
```
[the remark this answers is not in the document]
Comment (nagarajnetcentric) [reply apps/frontend/src/components/Tag/Tag.tsx]: Callback is triggered and updated set method

File: apps/frontend/src/components/Tag/Tag.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nagarajnetcentric) [reply apps/frontend/src/components/Tag/Tag.tsx]: This image is used to show cancel(X) icon inside button

File: apps/frontend/src/components/Tag/Tag.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    onClick();
+  }
+
+  return (
+    <button className={clsx(commonClasses,{
+        'flex bg-midnight-blue text-white': selected === true,
+        'text-midnight-blue border-midnight-blue hover:bg-midnight-blue hover:text-white': selected === false
+    })} onClick={handleClick}>
+      {label}
+      {selected && <img className='pl-[1.125rem] pt-[0.313rem]' src={vector} alt="cancel" />}
```
Comment (alexiscoelho) [inline apps/frontend/src/components/Tag/Tag.tsx:27]: `pl-0.5` it is the same as `pl-[1.125rem]`
`pt-1.5` it is the same as `pt-[0.313rem]`
Same for the above `px-[1.875rem] py-[0.938rem]` try to find the closest match:
https://tailwindcss.com/docs/padding

File: apps/frontend/src/components/Tag/Tag.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import vector from '../../assets/icons/vector.svg';
```
Comment (alexiscoelho) [inline apps/frontend/src/components/Tag/Tag.tsx:2]: Also, rename this vector to something meaningful. Like `cross` or something

**Review by alexiscoelho — CHANGES_REQUESTED** [review]

File: apps/frontend/src/components/Tag/Tag.tsx (hunk 8 above)
Comment (nagarajnetcentric) [reply apps/frontend/src/components/Tag/Tag.tsx:27]: No `pl-0.5` is equal to `pl-[0.125rem]`  not `1.125rem` and pt-1.5 is equal to `pt-[0.375rem]` not `pt-[0.313rem]`.

File: apps/frontend/src/components/Tag/Tag.tsx (hunk 9 above)
Comment (nagarajnetcentric) [reply apps/frontend/src/components/Tag/Tag.tsx:2]: will rename it to cross

File: apps/frontend/src/components/Tag/Tag.tsx (hunk 8 above)
Comment (alexiscoelho) [reply apps/frontend/src/components/Tag/Tag.tsx]: pl-5 then, the closest

**Review by alexiscoelho — APPROVED** [review]

**Pull request merged**